README


This archive contains the distribution DateTime-Format-Strptime,
version 1.55:

  Parse and format strp and strf time patterns

This software is Copyright (c) 2014 by Dave Rolsky.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)


This README file was generated by Dist::Zilla::Plugin::Readme v5.015.



Changes
1.56     2014-08-07

- Recent DateTime::TimeZone changes broke the tests.

- Fixed Latin-1 content in test code. It's now UTF-8.


1.55     2014-08-07.

- Broken release. I deleted this from PAUSE.


1.54     2013-04-02

- Require DateTime.pm 1.00 because without it tests will break.


1.53     2013-04-02

- A fix in DateTime.pm 1.00 broke a test in this distro. Reported by Anthony J
  Lucas. RT #84371.


1.52     2012-07-01

- Shut up "unescaped braces in regex" warning from 5.17.0. RT #77514. Patch by
  Zefram.


1.51     2012-05-27

- Packaging cleanup, including listing Test::More as a test prereq, not a
  runtime prereq. RT #76128.


1.5000   2010-10-16

- This module did not recognize UTC as a valid time zone. Patch by Danijel
  Tašov. RT #59209.


1.4000   2010-06-28

- Actually update $VERSION in module file. Reported by David Wheeler.


1.3000   2010-06-26

- Specifiers which allowed for leading space before a number (like %e) would
  cause DateTime.pm to throw an error if the date being parsed actually
  contained leading space. Patch by Alex Vandiver. RT #58459.

- License is now Artistic 2.0


1.2000   2010-03-19

- Updated to use non-deprecated DateTime::Locale API


1.1000   2009-07-13 -- Regex Pattern

- If the pattern you pass in is a regular expression, that will be honored.

- Changed the locale tests to use require the latest version of Locale until
  the target stops moving.


1.0901   2009-05-16 -- Official release of 1.0900_01

1.0900_01 2009-04-18

- Makefile.PL changes for Windows users as per issue #16


1.0900   2009-02-22

- It seems that I also wasn't seeing notifications from RT (please don't use
  it, use the Gooogle project) so all the following are fixed:

    - 36672   Started failing mid May
    - 23313   Bug handling time zones like America/New_York
    - 25555   Module dies even when on_error is 'undef'
    - 23768   Olson timezone handling incorrect
    - 22450   locale test failing with bleadperl
    - 20487   nmake test_more fail (with patch); incorrect META.yml
    - 12071   format_datetime uses datetime locale rather than format locale
    - 11863   bug in DateTime::Format::Strptime 1.0601 when using %s

  - And a couple from Google:
    - #8      Add DateTime::Locale to documentation
    - #10     Parsing bug -- can't detect word boundry after month abbr

1.0800   2008-08-07

- It seems that I wasn't getting notifications from Google when people had
  reported bugs, so there's a pile of fixes in this release. Hopefully that
  fixes everyone's issues.

1.0702   2007-09-19

- Updated the META.yml file to have the correct 'resources' and match the
  latest spec

- Updated the docus with a 'resources' section that has the same information
  in it for readers of POD.

- This version ONLY changes the documentation and so it not a required update.


1.0701   2007-09-18

- Many people pointed out that while this module hadn't broken the tests for
  the French locale had. This is due to a new source for the data in
  DateTime::Locale.

- This version ONLY changes the tests and so it not a required update.


1.0700 Sat,  5 Nov 2005 09:44:10 +1100

- Mike Schilli pointed out that strings without time zones or constructors
  without a time zone should be returning a DateTime in the floating time zone
  rather than UTC.

- Jason Bodnar requested greater allowance for time zones in strings .. so
  I've now added the ability to use an Olson time zone identifier with
  %O. Note that this is a token specifically added to Strptime and it WILL NOT
  WORK with DateTime's strftime method.


1.0601 Wed,  1 Sep 2004 07:52:44 +1000

- Dave Faraldo and Jonathan Lefter pointed out that one of the new Locale
  tests in t/006... will fail on the 30th and 31st of the month as not all
  months have those days.  Patch supplied by Jonathan has been applied.

- This is just a test fix and doesn't alter the way the module runs in any
  way. If you already got 1.06 to run then you don't need this.


1.0600 Sat, 28 Aug 2004 15:02:47 +1000

- Fixed bug from RT (#7502) from dfaraldo@redhat.com that made validation fall
  over by assuming midnight always exists. The patch now assumes Midday exists
  (both he and I assume that DST will never start at midday anywhere anytime!)

- This is a major change and you should install this release if you ever use
  any time_zone other than floating or UTC. And if you don't use them today,
  you may tomorrow - so upgrade OK?


1.0500 Wed, 18 Aug 2004 17:24:32 +1000

- Adapted and applied patches from Jean Forget to allow day names and month
  names to have \W characters and to allow am/pm values from other locales

- Jean's patch also included some doc patches

- Patched the synopsis a Jean suggestion to demonstrate how to use the
  non-default error behaviors

- Added tests for these bugs

- Added t/more/* and the make test_more target so I can test every locale if I
  want to.

- It's over a year since I deprecated the language parameter so I've now
  removed it. You've had a year of warnings!

- This is a major change and you should install this release if you use any
  locale other than en.


1.0400 Sun, 10 Aug 2003 00:22:00 +1000

- Applied patches from Joshua Hoblitt to move the the brand new
  DateTime::Locale rather than the old ::Language modules

- Implemented %x, %X and %c for locale formats

- Fixed a bug on the two digit year determinator

- Added a test for locales known as 004locale.t

- This is a major change and you should install this release if you are using
  DateTime >= 0.14 (the first release with Locale)


1.0302 Sat, 28 Jun 2003 09:15:21 +1000

- Test 004 was failing on Windows due to the path delimiter being a backslash
  rather than a slash. This release should also fix Strptime for Mac users,
  although I've had no complaints from such users. (Myself being one of the
  only ones currently using Strptime AFAIK!) Thanks Ron Hill for the bug
  report.


1.0301 Wed, 25 Jun 2003 22:40:12 +1000


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;



use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Parse and format strp and strf time patterns",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>, Rick Measham <rickm\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "DateTime-Format-Strptime",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "DateTime::Format::Strptime",
  "PREREQ_PM" => {
    "Carp" => 0,
    "DateTime" => "1.00",
    "DateTime::Locale" => "0.45",
    "DateTime::TimeZone" => "0.79",
    "Exporter" => 0,
    "Params::Validate" => "0.64",
    "strict" => 0,
    "vars" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88",
    "utf8" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.56",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "DateTime" => "1.00",
  "DateTime::Locale" => "0.45",
  "DateTime::TimeZone" => "0.79",
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Params::Validate" => "0.64",
  "Test::More" => "0.88",
  "strict" => 0,
  "utf8" => 0,
  "vars" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



